pandas/core/indexes/accessors.py

```diff
@@ -396,6 +396,53 @@ def freq(self):
         '2YS-JAN'
         """
         return self._get_values().inferred_freq
+    
+    @property
+    def date(self):
```

Comment:
Instead of adding a method, I think all we need to do is add `dtype=result.dtype` to the last line of `Properties._delegate_property_get` in this file.